adoc/deployment-bootstrap.adoc

```diff
@@ -389,6 +390,144 @@ Clusters provisioned following link:{docurl}single-html/caasp-deployment/#_using
+
+  function setup {
+    NAME=$1
+    echo "[$NAME]"
+    govc vm.power -dc=$DATACENTER -off $NAME
+    govc vm.change -dc=$DATACENTER -vm=$NAME -e="disk.enableUUID=1" &&\
+      echo "Configured disk.enabledUUID: 1"
+    govc vm.power -dc=$DATACENTER -on $NAME
+  }
+
+  for vm in ${VMS[@]}
+  do
+    setup $vm &
+  done
+  wait
+====
++
+<1> The datacenter where cluster virtual machines are in.
+<2> The cluster virtual machine names.
+
+. Update the provider ID for all Kuberentes nodes.
++
+====
+  DATACENTER="<VC_DATACENTER>" // <1>
+  CLUSTER_PREFIX="<VC_CLUSTER_PREFIX>" // <2>
+  for vm in $(govc ls "/$DATACENTER/vm/$CLUSTER_PREFIX-cluster")
+  do
+    VM_INFO=$(govc vm.info -json -dc=$DATACENTER -vm.ipath="/$vm" -e=true)
+    VM_NAME=$(jq -r ' .VirtualMachines[] | .Name' <<< $VM_INFO)
+    [[ $VM_NAME == *"-lb-"* ]] && continue
+    VM_UUID=$( jq -r ' .VirtualMachines[] | .Config.Uuid' <<< $VM_INFO )
+    echo "Patching $VM_NAME with UUID:$VM_UUID"
+    kubectl patch node $VM_NAME -p "{\"spec\":{\"providerID\":\"vsphere://$VM_UUID\"}}"
+  done
+====
++
+<1> The datacenter where cluster virtual machines are in.
+<2> Prefix for all machines of the cluster.
+
+. Create /etc/kubernetes/vsphere.config in every master and worker nodes. Refer to <<vsphere-cloud-provider-configuration>> for details.
+
+. On local machine, save kubeadm-config as `kubeadm-config.conf`.
++
+====
+kubectl -n kube-system get cm/kubeadm-config -o yaml > kubeadm-config.conf
+====
+
+. Edit the `kubeadm-config.conf` to add cloud-provider and relate configurations.
++
+====
+  data:
+    ClusterConfiguration: |
+      apiServer:
+        extraArgs:
+          cloud-config: /etc/kubernetes/vsphere.conf
+          cloud-provider: vsphere
+        extraVolumes:
+        - hostPath: /etc/kubernetes/vsphere.conf
+          mountPath: /etc/kubernetes/vsphere.conf
+          name: cloud-config
+          pathType: FileOrCreate
+          readOnly: true
+      controllerManager:
+        extraArgs:
+          cloud-config: /etc/kubernetes/vsphere.conf
+          cloud-provider: vsphere
+        extraVolumes:
+        - hostPath: /etc/kubernetes/vsphere.conf
+          mountPath: /etc/kubernetes/vsphere.conf
+          name: cloud-config
+          pathType: FileOrCreate
+          readOnly: true
+====
+
+. Apply the kubeadm-config to the cluster.
++
+====
+kubectl apply -f kubeadm-config.conf
+====
+
+. On every master node, update kubelet.
++
+====
+  sudo systemctl stop kubelet
+  source /var/lib/kubelet/kubeadm-flags.env
+  echo KUBELET_KUBEADM_ARGS='"'--cloud-config=/etc/kubernetes/vsphere.conf --cloud-provider=vsphere $KUBELET_KUBEADM_ARGS'"' > /tmp/kubeadm-flags.env
+  sudo mv /tmp/kubeadm-flags.env /var/lib/kubelet/kubeadm-flags.env
+  sudo systemctl start kubelet
+====
+
+. On every master node, update control-plane components.
++
+====
+sudo kubeadm upgrade node phase control-plane --etcd-upgrade=false
+====
+
+. On every worker node, update kubelet.
```

Comment:
Is it duplicated as the previous step?

Reply:
Yes this is duplicated but one step in between. I've separated this intentionally so there is no mix procedures between masters and workers, and user doesn't have to switch back and forth.

Reply:
Thanks for the explanation!